packages/teleport/src/NotificationContext.tsx

```diff
@@ -0,0 +1,30 @@
+import { createContext, useState } from 'react';
+
+// Notification Context
+type NotificationContextType = ReturnType<typeof makeNotificationContextValue>;
+
+const NotificationContext = createContext<NotificationContextType | null>(null);
+
+// Context Value
+const makeNotificationContextValue = () => {
+  const [notifications, setNotifications] = useState([]);
+
+  const removeNotification = (index: number) => {
+    setNotifications([
+      ...notifications.slice(0, index),
+      ...notifications.slice(index + 1),
+    ]);
+  };
+
+  const addNotification = (message: string) => {
+    setNotifications([...notifications, message]);
```

Comment:
I think that `notifications` needs to be something like `Notification[]` instead of `string` type where...

```ts
type Level = "message" | "warning" | "alert";

type Notification = {
  message: string;
  id: number;
  level: Level;
};
```

or even

```ts
type Level = "message" | "warning" | "alert";

type NotificationList = {
  [id: string]: Notification;
};

type Notification = {
  message: string;
  level: Level;
}
```

This is to support having different colours of notification and to be able to remove each notification by ID instead of the index in the list as that index will change over time.